framework/src/materials/GenericConstant2DArray.C

```diff
@@ -25,6 +25,7 @@ GenericConstant2DArray::validParams()
   params.declareControllable("prop_value");
   params.addClassDescription(
       "A material evaluating one material property in type of RealEigenMatrix");
+  params.set<MooseEnum>("constant_on") = "SUBDOMAIN";
```

Comment:
Do these objects now only work with "SUBDOMAIN" or whatever you are setting here? If so, you should suppress the parameter so it cannot be changed.

Reply:
No, any other option would still work but jut be less efficient.